composables/cache.ts

```diff
@@ -26,23 +25,12 @@ export function fetchStatus(id: string, force = false): Promise<mastodon.v1.Stat
   if (cached && !force)
     return Promise.resolve(cached)
 
-  let fetchPromise = fetchPromises.get(key)
-  if (!fetchPromise || force) {
-    fetchPromise = useMastoClient().v1.statuses.$select(id).fetch()
-      .then((status) => {
-        cacheStatus(status)
-        return Promise.resolve(status)
-      })
-      .finally(() => fetchPromises.delete(key))
-    fetchPromises.set(key, fetchPromise)
-  }
-
-  const promise = new Promise<any>((resolve) => {
-    fetchPromise!.then(resolve)
-  })
-  if (!cache.has(key) || force)
-    cache.set(key, promise)
-
+  const promise = useMastoClient().v1.statuses.$select(id).fetch()
+    .then((status) => {
+      cacheStatus(status)
```

Comment:
This is a bit confusing. Is it really needed for the `key` to be recalculated? Couldn't we directly call `cache.set(key, status)` here?

Reply:
We register 2 things there... the code is the original,  we can chabge it but the code will be duplicated